test/editing.js

```diff
@@ -523,6 +524,137 @@ export default function () {
       );
     }, 10);
   });
+  it('paste data and fill it down/over', function (done) {
+    const grid = g({
+      test: this.test,
+      data: [
+        {
+          field1: 'Value 1',
+          field2: 'Value 2',
+          field3: 'Value 3',
+        },
+        {},
+        {},
+      ],
+    });
+
+    grid.focus();
+    grid.selectRow(0, false, false, false);
+    grid.selectRow(2, false, true, false);
+
+    grid.paste({
+      clipboardData: {
+        items: [
+          {
+            type: 'text/plain',
+            getAsString: function (callback) {
+              callback('New value');
+            },
+          },
+        ],
+      },
+    });
+
+    setTimeout(function () {
+      try {
+        doAssert(grid.viewData.length == 3, 'Should have 3 rows exactly');
+        doAssert(Object.keys(grid.viewData[0]).length == 3, 'Should have 3 columns exactly');
+
+        for (let i = 0; i < grid.viewData.length; i++) {
+          for (const columnKey in grid.viewData[i]) {
+            const currentValue = grid.viewData[i][columnKey];
+            doAssert(
+              currentValue === 'New value',
+              'Value for "' + columnKey + '" should be "New value", but got ' + currentValue
+            );
+          }
+        }
+
+        done();
+      } catch (error) {
+        done(error);
+      }
+    }, 10);
+  });
+  it('paste data with a custom fill down/over function', function (done) {
+    const grid = g({
+      test: this.test,
+      data: [
+        {
+          field1: 'Value 1',
+          field2: 'Value 2',
+          field3: 'Value 3',
+        },
+        {},
+      ],
+      fillDownAttributes: {
+        cellCallback: function (args) {
```

Comment:
I feel like I'm missing something, but: what are the _other_ properties of `fillDownAttributes` ? I'm only ever seeing `cellCallback` and no other properties, it seems?

Reply:
Yes, it seemed like it is a common in the source code, and I also thought `filling` may get additional attributes. Do you think that is unnecessary? I think I can rename it to something like `fillOverCallback` or something.

Reply:
True! I think the difference is that in the other cases there's more than one attribute being used, and I wonder if we (will) need more than just the callback function: I feel like all the logic is contained in that one function, which is under the user's control. There _might_ be an argument to be made for parameters to control how/when the fill function should be called, but I cannot think of any, and I don't see the use case for it right now in this PR.

Any ideas?

Reply:
I think, for the sake of compatibility, using an umbrella attribute for the filling behavior might be a good idea. Right now, `cellCallback` is invoked for the filling of each cell, hence the name. But in the future, we might want to add a function that completely overrides the fill behavior, for instance.

Btw, cell-based callback might sound like we have less flexibility, but it is simpler to work with, and I think `args` provides all the necessary input for manipulating the data easily.

Reply:
> Btw, cell-based callback might sound like we have less flexibility, but it is simpler to work with, and I think args provides all the necessary input for manipulating the data easily.

I think I agree with this: the fill function should simply receive all the arguments such as cell position, and any other data it needs to determine the current cell's value.

> I think, for the sake of compatibility, using an umbrella attribute for the filling behavior might be a good idea.

This I'm not so sure about — compatibility with what, other arguments? In that case: I feel like we're in a process of gradually improving the public interface (hopefully not breaking anything in the process :), so some inconsistencies are allowed.

And for me, the fact that there are no other attributes than `cellCallback` is an indication that `fillAttributes` is not a great fit. Simpler for documentation and explaining the feature would be a single attribute, e.g. `fillCellFunction` or `fillCellCallback` — to me that feels immediately clear what its intent and shape is.

But I admit, it's a bit of a balancing act. If you feel strongly about `fillAttributes` I'll accept it, but I'm just not entirely convinced yet :-)

Reply:
Thanks for the reply :) I don't have a strong preference either, and as you said, so we can go with `fillCellCallback` since there isn't a certainty that `fillAttributes` will ever be expanded upon :)